src/features/Authentication/organisms/SignUpForm/SignUpForm.js

```diff
@@ -0,0 +1,125 @@
+    ),
+});
+
+const INITIAL_VALUES = {
+  firstName: '',
+  lastName: '',
+  email: '',
+  password: '',
+  passwordConfirmation: '',
+}
+
+const SignUpForm = () => {
+  const [{ fetching, error }, createUser] = useMutation(CREATE_USER);
+
+  const onSubmit = async (values, { setSubmitting }, {error}) => {
+    await createUser(values).then(result => {
+      if (result.error) {
+        console.error('Oh no!', result.error.message);
+      }
+    });
+    setSubmitting(false);
+  }
+
+  return (
+    <div>
+      <Typography variant="h3" component="div" gutterBottom align="center">
+        Sign Up
+      </Typography>
+      <Formik
+        initialValues={INITIAL_VALUES}
+        validationSchema={SIGNUP_SCHEMA}
+        onSubmit={(values, setSubmitting) => {
+          onSubmit(values, setSubmitting, {error});
+        }}
+      >
+        {({ isSubmitting}) => (
+          <Grid item xs={12} sm={8} md={5}>
+            <Box
+              sx={{
+                my: 8,
+                mx: 4,
+                display: 'flex',
+                flexDirection: 'column',
+                alignItems: 'center',
+              }}
+            >
+              {error &&
+                <Grid container spacing={2} sx={{ mt: 3, mb: 3 }}>
+                  <Grid item xs={12}>
+                    <Alert severity="error" align='left'>
+                      <AlertTitle>Error</AlertTitle>
+                      {error.message}
+                    </Alert>
+                  </Grid>
+                </Grid>
+              }
+              <Form>
+                <Grid container spacing={2}>
+                  <Grid item xs={12}>
+                    <Field component={TextField} name="firstName" fullWidth label="First name" placeholder="First name" variant="outlined" disabled={isSubmitting}/>
+                  </Grid>
+                  <Grid item xs={12}>
+                    <Field component={TextField} name="lastName" fullWidth label="Last name" placeholder="Last name" variant="outlined" disabled={isSubmitting}/>
+                  </Grid>
+                  <Grid item xs={12}>
+                    <Field component={TextField} name="email" type="email" fullWidth label="Email" placeholder="email@mail.com" variant="outlined" disabled={isSubmitting}/>
+                  </Grid>
+                  <Grid item xs={12}>
+                    <Field component={TextField} name="password" type="password" fullWidth label="Password" placeholder="Password" variant="outlined" disabled={isSubmitting} />
+                  </Grid>
+                  <Grid item xs={12}>
+                    <Button
+                      type="submit"
+                      fullWidth
+                      variant="contained"
+                      sx={{ mt: 3, mb: 2 }}
+                      disabled={isSubmitting}
+                    >
+                      Sign In {isSubmitting && fetching && <CircularProgress color="inherit" /> }
```

Comment:
fetching would be enough

Reply:
disabled={isSubmitting} and for CircularProgress ?

Reply:
```isSubmitting && fetching && <CircularProgress... ``` -> ```fetching && <CircularProgress... ```